tests/endpoints/test_forms.py

```diff
@@ -41,3 +45,121 @@ def test_get__ok(self):
                     form_id=fixture["response_data"][0]["xmlFormId"]
                 )
                 self.assertIsInstance(observed, Form)
+
+    def test_update__requires_definition_or_attachments(self):
+        with Client() as client:
+            try:
+                client.forms.update("foo")
+            except PyODKError:
+                pass
+            else:
+                self.fail("Exception expected")
+
+    @patch("pyodk._endpoints.forms.FormDraftAttachmentService")
+    @patch("pyodk._endpoints.forms.FormDraftService")
+    def test_update__definition_only__creates_then_publishes_draft(
```

Comment:
I've re-written it using a helper function so that these mock tests can be more succinct.